runtime/service/src/main/java/org/apache/polaris/service/auth/PolarisCredential.java

```diff
@@ -31,14 +31,22 @@
 public interface PolarisCredential extends Credential {
 
   static PolarisCredential of(
-      @Nullable Long principalId, @Nullable String principalName, Set<String> principalRoles) {
```

Comment:
Generally speaking, it is preferable to add new builder methods for new optional parameters (and chain old methods into them). This way existing code that does not use new optional parameter won't have to change.